controller/controller.go

```diff
@@ -1287,6 +1323,19 @@ func (ctrl *ProvisionController) provisionClaimOperation(claim *v1.PersistentVol
 		}
 		err = fmt.Errorf("failed to provision volume with StorageClass %q: %v", claimClass, err)
 		ctrl.eventRecorder.Event(claim, v1.EventTypeWarning, "ProvisioningFailed", err.Error())
+		if _, ok := claim.Annotations[annSelectedNode]; ok && result == ProvisioningFinished {
+			// For dynamic PV provisioning with delay binding, provisioner may fail
+			// because node is wrong (permanent error) or currently unusable (not enough
+			// capacity). In the later case, the provisioner must decide whether it wants
+			// to keep retrying (i.e. return ProvisioningNoChange) or give up on the
+			// node (i.e. return ProvisioningFinished). This code here only gives
+			// up on the node when it gets ProvisioningFinished as result.
```

Comment:
I was debating with myself whether I should reuse `ProvisioningFinished` here or introduce a new, more explicit `ProvisioningReschedule`. The problem with using `ProvisioningFinished` is that all final errors will then also trigger a rescheduling, which may not be the best course of action.

Suppose there's really something wrong with the PVC such that provisioning has to stop permanently. As it stands now, that no longer happens for volumes with delayed binding.

Another, related aspect is whether the external-provisioner should trigger a rescheduling immediately when receiving a `ResourceExhausted`. On the one hand, we want to try on a different node as quickly as possible. On the other hand, there's no exponential backoff anymore here if the volume really cannot be provisioned anywhere at the moment. Is there perhaps some delay built into the pod scheduler?

Reply:
IMO, there should be separate `ProvisioningReschedule`. A Pod/PVC should not hop around nodes with each failure e.g. in provisioning credentials.

Reply:
I tend to agree. Let me change it accordingly.